docker/nginx.conf

```diff
@@ -0,0 +1,217 @@
+  gzip_proxied any;
+  gzip_types
+    text/plain
+    text/css
+    text/js
+    text/xml
+    text/javascript
+    application/javascript
+    application/x-javascript
+    application/json
+    application/xml
+    application/rss+xml
+    application/wasm
+    image/svg+xml;
+  gzip_buffers 16 8k;
+
+  open_file_cache max=1000 inactive=20s;
+  open_file_cache_valid 30s;
+  open_file_cache_min_uses 5;
+  open_file_cache_errors off;
+
+  types {
+    application/wasm wasm;
+  }
+
+  map $http_upgrade $connection_upgrade {
+    default upgrade;
+    '' close;
+  }
+
+  map $sent_http_content_type $expires {
+    default                                 1M;
+
+    # No content
+    ""                                      off;
+
+    # CSS
+    ~*text/css                              1y;
+
+    # Data interchange
+    ~*application/atom\+xml                 1h;
+    ~*application/rdf\+xml                  1h;
+    ~*application/rss\+xml                  1h;
+
+    ~*application/json                      0;
+    ~*application/ld\+json                  0;
+    ~*application/schema\+json              0;
+    ~*application/geo\+json                 0;
+    ~*application/xml                       0;
+    ~*text/calendar                         0;
+    ~*text/xml                              0;
+
+    # Favicon (cannot be renamed!) and cursor images
+    ~*image/vnd.microsoft.icon              1w;
+    ~*image/x-icon                          1w;
+
+    # HTML
+    ~*text/html                             0;
+
+    # JavaScript
+    ~*application/javascript                1y;
+    ~*application/x-javascript              1y;
+    ~*text/javascript                       1y;
+
+    # Manifest files
+    ~*application/manifest\+json            1w;
+    ~*application/x-web-app-manifest\+json  0;
+    ~*text/cache-manifest                   0;
+
+    # Markdown
+    ~*text/markdown                         0;
+
+    # Media files
+    ~*audio/                                1M;
+    ~*image/                                1M;
+    ~*video/                                1M;
+
+    # WebAssembly
+    ~*application/wasm                      1y;
+
+    # Web fonts
+    ~*font/                                 1M;
+    ~*application/vnd.ms-fontobject         1M;
+    ~*application/x-font-ttf                1M;
+    ~*application/x-font-woff               1M;
+    ~*application/font-woff                 1M;
+    ~*application/font-woff2                1M;
+
+    # Other
+    ~*text/x-cross-domain-policy            1w;
+  }
+
+  server {
+    listen [::]:9999;
+    listen 9999;
+
+    server_name _;
+
+    # Path for static files
+    root /usr/share/nginx/html;
+
+    location /nginx_status {
+      stub_status;
+      allow 127.0.0.1;
+      deny all;
+    }
+
+    location / {
+      try_files $uri /index.html;
+    }
+
+    # NB: ALL CSP headers are in default because otherwise for each type of source you need to fully specify allowed domains
```

Comment:
should we remove these for this image in case community members want to use it? seems pretty specific to what we are doing. we can use an `include` directive like we do for ingress rules... so it'd look like "include(./csp.conf)" or something, and we copy that file into the environment before building in monitor-ci

Reply:
Can we sync on this just so I'm a little clearer on the implementation?